sdk/apis/roundtrip_test.go

```diff
@@ -131,6 +142,9 @@ func testGenericConversion[V1 runtime.Object, V2 runtime.Object](
 		err = toV1(intermediate, result, nil)
 		require.NoError(t, err)
 
+		// remove annotations that we expect on the object due to conversion data stored.
+		removeAnnotations(t, result, v1ExpectedAnnotations)
```

Comment:
Nitpicky but `removeAnnotations` and `expectedAnnotation` reads antithetical to me. I'd expect `expectedAnnotations` to be the annotations I expect on the final object.

Maye `v1TrimAnnotations` to explicitly say that these will be removed?

Reply:
the idea here was that for the purpose of comparison, we "expect" this annotation to be there, so we need to remove it before checking equality. But I don't have strong feelings about renaming it.

Reply:
Although it could be argued that if the annotation is missing that the conversion overall failed and that the test will fail anyhow.

Reply:
I'd also extend the comment to clarify this properly, at the moment, it's not very clear out from the comment alone.